### PR TITLE
xcodegen: init at 2.42.0

### LINKED ISSUE #208816: XcodeGen
**Project description**
_describe the project a little_

XcodeGen is a command line tool written in Swift that generates your Xcode project using your folder structure and a project spec.

**Metadata**

* homepage URL: https://github.com/yonaskolb/XcodeGen
* source URL: 
* license: mit
* platforms: darwin

### PR DESCRIPTION
**Pull request by Samasaur1**
Adds [XcodeGen](https://github.com/yonaskolb/XcodeGen). Closes #208816

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/xc/xcodegen/package.nix [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      # Now apply a patch
+      patch -p1 -d .build/checkouts/Spectre -i ${./0001-spectre-xct-record.patch}
+    '';
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp "$(swiftpmBinPath)/${pname}" $out/bin/
+  '';
+
+  meta = with lib; {
```
Comment (steeleduncan) [inline pkgs/by-name/xc/xcodegen/package.nix:55]: `with lib` is deprecated #208242

Comment (steeleduncan) [inline pkgs/by-name/xc/xcodegen/package.nix:55]: `mainProgram` is missing

**Review by steeleduncan — CHANGES_REQUESTED** [review]

**Comment by Samasaur1** [comment]
That failed eval is actually a git error, not sure what's going on there

**Review by GaetanLepage — CHANGES_REQUESTED** [review]

**Comment by Samasaur1** [comment]
I made the changes you suggested, but I have a question about `passthru.updateScript`. I was under the impression that the default was to use `nix-update-script`. Are you just suggesting that I be explicit here or am I wrong about the default behavior?

**Review by GaetanLepage — APPROVED** [review]
LGTM

**Review by DimitarNestorov — APPROVED** [review]
LGTM

Tested basic functionality with following commands, works fine

```sh
$ echo "name: Test" > project.yml
$ xcodegen --version
Version: 2.42.0
$ xcodegen
⚙️  Generating plists...
⚙️  Generating project...
No "base" settings found
No "debug config" settings found
No "release config" settings found
⚙️  Writing project...
Created project at /Users/dimitar/.cache/nixpkgs-review/pr-367199/Test.xcodeproj
```

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 367199`

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>xcodegen</li>
  </ul>
</details>

**Comment by DanielSincere** [comment]
Hello, love that xcodegen is supported here. However, it's not installed correctly. I'm getting the following errors when I run it, which results a different xcodeproj file when running xcodegen when installed by [homebrew](https://brew.sh) or[mint](https://github.com/yonaskolb/Mint). I cannot switch to Nix until this is resolved, as I rely on xcodegen.

Here is my error
```
Running `xcodegen generate -s ./xcodegen.yml`
⚙️  Generating plists...
⚙️  Generating project...
No "base" settings found
No "debug config" settings found
No "release config" settings found
No "iOS" settings found
⚙️  Writing project...
```

I see that @DimitarNestorov received a similar error [here](https://github.com/NixOS/nixpkgs/pull/367199#issuecomment-2564647131).

The missing files are static settings files packaged along side the xcodegen binary, from the `SettingsPresets` folder in [source code](https://github.com/yonaskolb/XcodeGen/tree/master/SettingPresets).

**Pull request merged**